stock_available_to_promise_release/models/stock_move.py

```diff
@@ -605,13 +644,22 @@ def unrelease(self, safe_unrelease=False):
         impacted_picking_ids = set()
 
         for move in moves_to_unrelease:
+            # When a move is returned, it is going straight to WH/Stock,
+            # skipping all intermediate zones (pick/pack).
+            # That is why we need to keep track of qty returned along the way.
+            # We do not want to return the same goods at each step.
+            # At a given step (pick/pack/ship), qty to return is
+            # move.product_uom_qty - cancelled_qty_at_step - already returned qties
+            qty_to_unrelease = move.product_uom_qty
```

Comment:
In the uom of the product as each move could have a different uom
```suggestion
            qty_to_unrelease = move.product_qty
```